### PR TITLE
linked routing between chapter and subchapter, added axios calls to s…

### PR DESCRIPTION
**Pull request by WiselyKwek**
linked routing between chapter and subchapter
added axios calls to server to render chapter and subchapter data from MongoDB

### CODE REVIEW

File: SPARK-App/src/pages/Chapters.jsx [hunk 1]
```diff
@@ -6,47 +6,62 @@ import AddIcon from '@mui/icons-material/Add'
...
+    //         title: "Chapter 3",
+    //         description: "This is the third chapter",
+    //         image: "../../src/handbook1.jpg"
+    //     },
+    //     {
+    //         title: "Chapter 3",
+    //         description: "This is the third chapter",
+    //         image: "../../src/handbook1.jpg"
+    //     },
+    // ]
```
Comment (peishan31) [inline SPARK-App/src/pages/Chapters.jsx]: can consider deleting commented codes

File: SPARK-App/src/components/chapters/ChapterCard.jsx [hunk 2]
```diff
@@ -4,32 +4,41 @@ import CardContent from '@mui/material/CardContent';
...
+                            { state: { parentChapterId: currentChapterId } })
+                    }
+                }>
+                {/* <CardMedia
                     component="img"
                     height="225"
                     image="../../assets/handbook1.jpg"
                     alt="green iguana"
-                />
+                /> */}
```
Comment (peishan31) [inline SPARK-App/src/components/chapters/ChapterCard.jsx]: can delete commented code

File: SPARK-App/src/pages/Chapters.jsx [hunk 3]
```diff
@@ -6,47 +6,62 @@ import AddIcon from '@mui/icons-material/Add'
...
+    useEffect(() => {
+        axios.get('http://localhost:8080/chapters')
+            .then(res => {
+                setChapters(res.data)
+            })
+            .catch(err => {
+                console.log(err)
+            })
+    }, [])
+    console.log(chapters)
```
Comment (peishan31) [inline SPARK-App/src/pages/Chapters.jsx]: delete console.log codes

File: SPARK-App/src/pages/Subchapters.jsx [hunk 4]
```diff
@@ -7,48 +7,38 @@ import AddIcon from '@mui/icons-material/Add'
...
-            image: "../../assets/handbook1.jpg"
-        },
-        {
-            title: "Subchapter 3",
-            description: "This is the third subchapter",
-            image: "../../assets/handbook1.jpg"
-        },
-    ]
+    const chapterId = location.state.parentChapterId
+    console.log(chapterId)
```
Comment (peishan31) [inline SPARK-App/src/pages/Subchapters.jsx]: delete console.log codes

File: SPARK-App/src/pages/Subchapters.jsx [hunk 5]
```diff
@@ -7,48 +7,38 @@ import AddIcon from '@mui/icons-material/Add'
...
+    useEffect(() => {
+        axios.get(`http://localhost:8080/chapters/${chapterId}/subchapters`)
+            .then(res => {
+                setSubchapters(res.data)
+            })
+            .catch(err => {
+                console.log(err)
+            })
+    }, [])
+    console.log(subchapters)
```
Comment (peishan31) [inline SPARK-App/src/pages/Subchapters.jsx]: delete console.log codes

File: Server/routes/chapterRouter.js [hunk 6]
```diff
@@ -53,10 +53,11 @@ chapterRouter.get('/:chapterId', async (req, res) => {
 chapterRouter.post("/", async (req, res) => {
     console.log("Create a chapter")
```
Comment (peishan31) [inline Server/routes/chapterRouter.js]: delete console.log codes

**Review by peishan31 — CHANGES_REQUESTED** [review]

File: SPARK-App/src/pages/Chapters.jsx (hunk 1 above)
Comment (WiselyKwek) [reply SPARK-App/src/pages/Chapters.jsx]: done

**Pull request merged**